### PR TITLE
Powheg ggHH_EWChL

### PR DESCRIPTION
**Pull request by MatthewDKnight**
These cards are for a resonant search for di-Higgs production in the HH->gamma gamma tau tau final state. The intention is to use the full Run 2 dataset so UL samples will be asked for all 3 years.

The card from the gridpack [1] used to make a similar dataset [2] in the bbgg final state was taken. Then the appropriate UL PDF changes were made. 

[1] /cvmfs/cms.cern.ch/phys_generator/gridpacks/slc6_amd64_gcc700/13TeV/powheg/V2/ggHH_EWChL_NNPDF31_13TeV_M125_cHHH1/v3/ggHH_EWChL_slc6_amd64_gcc700_CMSSW_10_2_5_patch1_my_ggHH_EWChL.tgz
[2] /GluGluToHHTo2B2G_node_cHHH1_TuneCP5_PSWeights_13TeV-powheg-pythia8/RunIIFall17MiniAODv2-PU2017_12Apr2018_94X_mc2017_realistic_v14-v1/MINIAODSIM

### CODE REVIEW

File: bin/Powheg/production/2017/13TeV/ggHH_EWChL/powheg_cHHH0.input [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+use-old-ubound  1    ! if 1 use norm of upper bounding function stored in pwgubound.dat, if present; <> 1 regenerate
+!
+ncall1  10000        ! number of calls for initializing the integration grid
+itmx1       3        ! number of iterations for initializing the integration grid
+ncall2  10000        ! number of calls for computing the integral and finding upper bound
+itmx2       3        ! number of iterations for computing the integral and finding upper bound
+foldcsi     1        ! number of folds on csi integration
+foldy       1        ! number of folds on  y  integration
+foldphi     1        ! number of folds on phi integration
+nubound 10000        ! number of bbarra calls to setup norm of upper bounding function
```
Comment (agrohsje) [inline bin/Powheg/production/2017/13TeV/ggHH_EWChL/powheg_cHHH0.input:48]: did you test that this is sufficient? what is the fraction of neg. weights?

File: bin/Powheg/production/2017/13TeV/ggHH_EWChL/powheg_cHHH0.input [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+fakevirt 1
+
+
+#pdfreweight 0
+
+#withnegweights 1 1
+rwl_group_events 2000
+lhapdf6maxsets 50
+rwl_file 'pwg-rwl.dat'
+rwl_format_rwgt 1
```
Comment (agrohsje) [inline bin/Powheg/production/2017/13TeV/ggHH_EWChL/powheg_cHHH0.input:118]: please remove.

File: bin/Powheg/production/2017/13TeV/ggHH_EWChL/powheg_cHHH0.input [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+#py8nompi 1
+
+check_bad_st1 1
+check_bad_st2 1
+max_io_bufsize  2000000
+
+
+
+
+fakevirt 1
```
Comment (agrohsje) [inline bin/Powheg/production/2017/13TeV/ggHH_EWChL/powheg_cHHH0.input:109]: did you check the impact?

**Comment by agrohsje** [comment]
Hi @MatthewDKnight. Do you mind having one template card and a small script that replaces the corresponding coupling?

File: bin/Powheg/production/2017/13TeV/ggHH_EWChL/powheg_cHHH0.input (hunk 1 above)
Comment (MatthewDKnight) [reply bin/Powheg/production/2017/13TeV/ggHH_EWChL/powheg_cHHH0.input:48]: The negative weight fraction for cHHH1 was 5.17e-2. Please let me know if that is low enough and if not, what fraction I should be aiming for.

File: bin/Powheg/production/2017/13TeV/ggHH_EWChL/powheg_cHHH0.input [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+! Born, plots and powheg strategy:
+#bornonly         1  ! (default 0) if 1 do Born only
+testplots         1  ! (default 0, do not) do NLO and PWHG distributions
+fastbtlbound      1  ! fast evaluation of Btilde upper bounds
+storemintupb      1  ! store function calls in binary file for more precise generation
+ubexcess_correct  1  ! corrects for error in upper bound estimate
+#storeinfo_rwgt 0
+softtest          0  ! no check of the soft limit in the real
+colltest          0  ! no check of the collinear limit in the real
```
[the remark this answers is not in the document]
Comment (MatthewDKnight) [reply bin/Powheg/production/2017/13TeV/ggHH_EWChL/powheg_cHHH0.input:69]: I took these cards from an equivalent re-reco dataset. I do not know why these options was chosen originally and I'm also not sure what their impact would be. Is there any chance that turning them back on could make things worse? If not, and the default is to have them on, I'd suggest turning them back on.

File: bin/Powheg/production/2017/13TeV/ggHH_EWChL/powheg_cHHH0.input (hunk 3 above)
Comment (MatthewDKnight) [reply bin/Powheg/production/2017/13TeV/ggHH_EWChL/powheg_cHHH0.input:109]: I have not checked the impact. If this check is necessary I'd hope it was done also for the re-reco dataset. However, I could not find the cards in genproductions for that dataset, meaning I can't check if there was a similar discussion, nor can I find who made the re-reco gridpack.

How would I go about checking the impact? Would I have to regenerate a gridpack with fakevirt=0 and then check the distributions?

File: bin/Powheg/production/2017/13TeV/ggHH_EWChL/powheg_cHHH0.input (hunk 2 above)
Comment (MatthewDKnight) [reply bin/Powheg/production/2017/13TeV/ggHH_EWChL/powheg_cHHH0.input:118]: I'll remove this when I make the template card + small script.

**Comment by MatthewDKnight** [comment]
No problem, I will do that.

**Pull request merged**